EIPS/eip-00.md

```diff
@@ -96,6 +97,28 @@ If `paymasterData` is specified, its first 20 bytes contain the address of a `pa
 
 If `deployerData` is specified, its first 20 bytes contain the address of a `deployer` contract.
 
+### No-op transaction subtype
```

Comment:
why is it better than prefix+counter ?
a prefix+counter can be extended to have an aggregator
So a prefix-counter (with empty aggregator) is what we need now, and when we fully define aggregation, we add prefix-counter-aggregator.

The important thing is use the same mechanism now, instead of changing mechanism when we introduce aggregator.

Reply:
In aggregation there is some execution and state change in the header transaction, but here the header would be just as much of a no-op transaction, just a required one and before the batch instead of after it.

Reply:
When adding support for aggregation, we'd need a lot more than just prefix+counter+aggregator. We also have state changes in this transaction. So I'm not sure how close these mechanisms are. I think that a no-op and a prefix+counter is very similar right now.

Reply:
currently, we can implement either prefix+counter, or suffix. There is no real reason to implement one or the other.
Looking forward, if we ever add aggregation, we will have to have prefix+counter (+aggregator).
So if we now add prefix+counter, then we have it "for free" when we do aggregation.
if we add now suffix, then it will have to be changed (or worse, add both mechanisms) when we add aggregation.

So I'm saying we need some good argument to add a new "suffix" transaction type now.

Reply:
I think it makes sense to use prefix+counter in a way that can be later extended for aggregation, since we already know we'll need that counter then.  Although the NOP below is compelling since it's really minimal.

Another question: if we add a prefix+counter should it be mandatory, or can we make it optional and revert to the current algorithm if we encounter type4 transactions without the prefix?  This prefix will only be needed when concatenating separate batches, or in aggregation.

Reply:
> Another question: if we add a prefix+counter should it be mandatory, or can we make it optional and revert to the current algorithm if we encounter type4 transactions without the prefix? This prefix will only be needed when concatenating separate batches, or in aggregation.

We can make it optional so in most cases it won't even be used, I think. A small bit of "over-engineering" but also makes it cleaner for the most common use-case.